federation/pkg/federatedtypes/scheduling.go

```diff
@@ -87,6 +93,57 @@ func (a *replicaSchedulingAdapter) IsSchedulingAdapter() bool {
 	return true
 }
 
+func isSelected(names []string, name string) bool {
```

Comment:
Rather replace with a simple set inclusion operation?  That would be less code, and more efficient.  I think we have a function in the utils library to do that.  StringSet or similar.